File: backend/app/models/review.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from app.database import Base

from sqlalchemy.orm import relationship




class Review(Base):
    __tablename__ = "reviews"
    __table_args__ = {"extend_existing": True}  # <--- Ajoutez cette ligne
    id = Column(Integer, primary_key=True, index=True)
    rating = Column(Float)
    comment = Column(String)
    user_id = Column(Integer, ForeignKey("users.id"))
    restaurant_id = Column(Integer, ForeignKey("restaurants.id"))

    restaurant = relationship("Restaurant", back_populates="reviews")

File: frontend/src/app/models/review.ts
export interface Review {
  id: number;
  restaurant_id: number;
  user_id: number;
  rating: number;
  comment: string | null;
  created_at: string | null;
}

export interface ReviewCreateRequest {
  restaurant_id: number;
  rating: number;
  comment?: string | null;
}
